jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/CheckForCollidingPaths.java

```diff
@@ -128,4 +129,8 @@ private static Pattern createPathPattern(String path) {
         return Pattern.compile(path);
     }
 
+    private static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor) {
```

Comment:
I do think it would be easier to read if not in specified as a function

Reply:
You mean inline it? I think distinctByKey says what it does.

Reply:
I did not see that the method actually kept state  and works as a filter with state. In that case its easier with a function as here. The method is highly generic,  The T could be swapped out for the class we are working on and the variable t could be called resource to enhance readability, if the intent is not to create a framework method that requires that flexibility

Reply:
Its just a note. We dont need to change, but the statething and the generic part together makes it a bit harder to see what is what